### ISSUE TITLE
Implement partner service for new campaign creation completed signal

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#8560]
## Feature Description

When the campaign creation portion of the PAX setup flow is completed, the partner `campaignService` can be used to respond to the event.

The handler is [defined as part of the service](https://www.npmjs.com/package/google-pax-sdk?activeTab=readme#:~:text=%2C%0A%7D%3B-,Campaign%20service,-let%20campaignService%20%3D) and provided to the app at launch time.

This handler should be contained within the `SetupMainPAX` component given that it is only ever relative to PAX onboarding/setup flow. See #8558 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The `SetupMainPAX` component should be updated to provide an implementation of the app's `notifyNewCampaignCreated` handler of the `campaignService`
	* This requires an update to the `PAX` component as well such that it should receive a new optional `onCampaignCreated` function prop
		- When provided, this function should be included as the value to the `services.campaignService.notifyNewCampaignCreated` definition ([example](https://github.com/google/partner-ads-experience/tree/main?tab=readme-ov-file#campaign-service))
	* The `onCampaignCreated` function passed should (when called) set the conversion ID and customer IDs currently being set by the main submit button
	* The main submit button should then only be enabled when these IDs have valid values in state, and the campaign creation has been completed
		- Invoking it should then submit changes / save and navigate as usual
	* The embedded app can also be hidden once the conditions are met to enable the main submit button as above although this may not be necessary depending on the behavior of the `displayMode`

## Implementation Brief

* [x] Within the `<SetupMainPAX>` component file:
	* Define a function which uses the `app` instance (from the existing value provided to `onLaunch` defined in #8557) and its available services (via `app.getServices()`) as should already be done in the component, but attached to the main submit button
		- Extract the existing behavior (setting conversion + customer IDs) except for the submit changes action from the main submit button to this new function as defined in the AC
		- Hide the main PAX component if necessary
		- Pass this function to the (new) `onCampaignCreated` prop of the `PAXEmbeddedApp`
* [x] Update the `PAXEmbeddedApp` to accept an optional `onCampaignCreated` prop
	- If provided a function, merge it into the definition of `services` provided to the app as `services.campaignService.notifyNewCampaignCreated: onCampaignCreated`

### Test Coverage

* Add or update existing tests for the `PAXEmbeddedApp` component to test that the callback function is included in the service definitions as expected

## QA Brief

* Set up Site Kit with `adsPax` feature flags
* Go to Ads module setup screen and click on `create account` button
  * If you do not have `adwords` scope granted, you should see notice bellow the button, upon clicking the button, you will be redirected to the OAuth flow
  * If you have the scope already, it should skip OAuth step and launch PAX app
* Upon completing all steps in PAX, a `complete setup` button should replace it
* Once clicked, it should connect Ads module and redirect you to the dashboard with success subtle banner showing
* Settings view of Ads module should show `Conversion Tracking ID` and `Customer ID` values, but not when editing the settings 

## Changelog entry

* Implement partner service for new campaign creation completed signal.

**Comment by aaemnnosttv** [comment]
Thanks @10upsimon – the signal here makes sense to trigger the sync of the relevant values into state (e.g. `setPaxConversionID(...)`), however the actual action to save (submit changes) should be handled from the `SetupMainPAX` component. So we might need to set a separate flag in the datastore (maybe `core/forms`) to indicate that campaign setup is complete, at which point we can respond by hiding the app (if needed) and potentially even triggering the save automatically, but again, that should live in the setup component I think. 

That's all getting into IB territory but useful context – the AC is basically there, but I think we probably just want to hold on calling for settings to be saved as something that this service is responsible for, while also keeping in mind that there could be value in storing a flag that campaign setup is completed in state. We don't need to call for requiring things that we don't know are going to be needed yet but it seems reasonable that we very well may need it.

**Comment by 10upsimon** [comment]
@aaemnnosttv this is ready for a re review

**Comment by aaemnnosttv** [comment]
Tweaked a bit but this one should be pretty straightforward now that it's following after #8558

IB ✅

**Comment by wpdarren** [comment]
### QA Update: ⚠️ 

@zutigrm I have a few observations based on the QAB:

> Upon completing all steps in PAX, a complete setup button should replace it

1. The `Complete setup` button is active throughout the setup steps, should it not be inactive until you complete the campaign? Users, be users, are likely to click it. 

3. When you've created the campaign performance, campaign settings widget appears.  From reading the QAB, I am not sure if this is expected or not, or, if we should just see the button to click.  I suspect users are not going to click the  Complete setup button it looks like it is setup. It's not obvious that you have to click the button to complete the setup. 

![image](https://github.com/google/site-kit-wp/assets/73545194/bdb3c8ce-ce81-4b28-8d7d-2754ab902790)

> Once clicked, it should connect Ads module and redirect you to the dashboard with success subtle banner showing.

1. When clicking the Complete setup CTA, at the moment, the Ads Conversion Tracking ID success message appears, I am assuming that this is expected, since we have #8661 for the next sprint.  

2. After loading the success notification, the page automatically scrolls to the bottom of the dashboard, underneath the monetization section. The user then has to scroll up the page which isn't very user-friendly. 

Interested in your thoughts here.

**Comment by wpdarren** [comment]
### QA Update: ⚠️ 

@zutigrm now that I am testing on the right branch 🤣 I just have two observations. 

- When I click on the **Complete setup** button, I am taken to the dashboard, the success notification appears, but after a few seconds the page automatically scrolls to the butotm underneath the Monetization area.  Maybe this is to do with the reporting widget that we've introduced on to the dshboard, but thought I would mention it. 
- The final point on the QAB:  `Settings view of Ads module should show Conversion Tracking ID and Customer ID values, but not when editing the settings` When I click on the edit link I do see the two values there. Is tihis an issue? 

**Verified:**

- When on the Ads module setup screen and I click on create account button and do not have adwords scope granted, I see notice bellow the button, upon clicking the button, you will be redirected to the OAuth flow.
- When on the Ads module setup screen and I have the scope already, it skips the OAuth step and launch PAX app
- Upon completing all steps in PAX, a complete setup button replaces it and once clicked, it connects the Ads module and redirects me to the dashboard with success subtle banner showing.

<details>
 <summary>Screenshots</summary>

![image](https://github.com/google/site-kit-wp/assets/73545194/b126b6f8-154a-4c21-9916-1aca529a4c93)
![image](https://github.com/google/site-kit-wp/assets/73545194/f055f18f-079a-4112-9550-5d33b18197f5)
![image](https://github.com/google/site-kit-wp/assets/73545194/5f8cd434-c49e-4aa2-b45c-c880939f7e68)
![image](https://github.com/google/site-kit-wp/assets/73545194/9daf0072-3e52-4202-ac88-6a09649af762)
![image](https://github.com/google/site-kit-wp/assets/73545194/9f822463-3649-4755-b359-f1585408ce6e)

</details>

**Comment by zutigrm** [comment]
Thanks @wpdarren 

> When I click on the Complete setup button, I am taken to the dashboard, the success notification appears, but after a few seconds the page automatically scrolls to the button underneath the Monetisation area. Maybe this is to do with the reporting widget that we've introduced on to the dashboard, but thought I would mention it.

I just noticed this behaviour while working on another issue, and was trying to inspect this. So far I verified that once reporting widget is shown it jumps to the bottom of the page. I am trying to debug this and determine if there is something we can do to prevent this, and also consult with @aaemnnosttv on his opinion. Sine this issue is about complete signal and redirection, we should probably address this in different issue as it is not in the scope of the current one.

> The final point on the QAB: Settings view of Ads module should show Conversion Tracking ID and Customer ID values, but not when editing the settings When I click on the edit link I do see the two values there. Is this an issue?

It was more in terms of, settings shouldn't be editable, with no fields allowing you to change them. The previous implementation did include the view-only values on edit view. This is issue focuses on saving the forwarded info, the showing of the fields was scoped and implemented in different issues. So what you see is fine

**Comment by wpdarren** [comment]
QA Update: ✅ 

**Verified:**

- When on the Ads module setup screen and I click on create account button and do not have adwords scope granted, I see notice bellow the button, upon clicking the button, I am redirected to the OAuth flow.
- When on the Ads module setup screen and I have the scope already, it skips the OAuth step and launch PAX app
- Upon completing all steps in PAX, a complete setup button replaces it and once clicked, it connects the Ads module and redirects me to the dashboard with success subtle banner showing.
- Settings view of Ads module shows Conversion Tracking ID and Customer ID values, but the user is not able to edit them. The edit link does work but no form appears allowing the users to edit. 

There is an automatic scroll when the user is redirected back to the main dashboard. We think this is because of the PAX widget, so is out of scope for this ticket. 

<details>
 <summary>Screenshots</summary>

![image](https://github.com/google/site-kit-wp/assets/73545194/b126b6f8-154a-4c21-9916-1aca529a4c93)
![image](https://github.com/google/site-kit-wp/assets/73545194/f055f18f-079a-4112-9550-5d33b18197f5)
![image](https://github.com/google/site-kit-wp/assets/73545194/5f8cd434-c49e-4aa2-b45c-c880939f7e68)
![image](https://github.com/google/site-kit-wp/assets/73545194/9daf0072-3e52-4202-ac88-6a09649af762)
![image](https://github.com/google/site-kit-wp/assets/73545194/9f822463-3649-4755-b359-f1585408ce6e)

</details>
